ksql-engine/src/main/java/io/confluent/ksql/util/KafkaTopicClientImpl.java

```diff
@@ -107,11 +127,62 @@ public void deleteTopics(List<String> topicsToDelete) {
       }
     }
     if (!failList.isEmpty()) {
-      throw new KsqlException("Failed to clean up topics: " + failList.stream().collect(Collectors
-                                                                                          .joining(",")));
+      throw new KsqlException("Failed to clean up topics: " + failList.stream()
+          .collect(Collectors.joining(",")));
+    }
+  }
+
+  @Override
+  public void deleteInternalTopics(String applicationId) {
+    if (!isDeleteTopicEnabled) {
+      log.info("Cannot delete topics since 'delete.topic.enable' is false. ");
+      return;
+    }
+    Set<String> topicNames = listTopicNames();
+    List<String> internalTopics = new ArrayList<>();
+    for (String topicName: topicNames) {
+      if (isInternalTopic(topicName, applicationId)) {
+        internalTopics.add(topicName);
+      }
+    }
+    if (!internalTopics.isEmpty()) {
+      deleteTopics(internalTopics);
     }
   }
 
+  private void init() throws ExecutionException, InterruptedException {
+    DescribeClusterResult describeClusterResult = adminClient.describeCluster();
+    this.nodes = new ArrayList<>(describeClusterResult.nodes().get());
+    if (!nodes.isEmpty()) {
+      ConfigResource resource = new ConfigResource(ConfigResource.Type.BROKER,
+                                                   String.valueOf(nodes.get(0).id()));
+      DescribeConfigsResult
+          describeConfigsResult = adminClient.describeConfigs(Collections.singleton(resource));
+      this.config = describeConfigsResult.all().get();
+      List<ConfigEntry> configEntries = config.get(resource)
+          .entries()
+          .stream()
+          .filter(configEntry -> configEntry.name().equalsIgnoreCase("delete.topic.enable"))
+          .collect(Collectors.toList());
+      if (!configEntries.isEmpty()) {
+        if (configEntries.get(0).value().equalsIgnoreCase("true")) {
+          this.isDeleteTopicEnabled = true;
+        } else {
+          this.isDeleteTopicEnabled = false;
+        }
+      }
+    } else {
+      throw new KsqlException("Could not fetch broker information. KSQL cannot initialize "
```

Comment:
i wonder if this should be an exception at this point? Perhaps this could be lazily initialized? i.e., it might be that the nodes/cluster are temporarily unavailable? In which case we maybe want to have a retry/backoff loop - @bluemonk3y WDYT?

Reply:
again - for now I would suggest we have an acceptable default - and ticket the desirable behaviour. Something at a higher level should own the problem of the retry - depending on the caller. i.e. Running embedded might want to fail fast and delegate responsibility, run in a cluster, might want the cluster manager to do that for me. Running a RestEndpoint, probably want retry w logging

Reply:
Agree with Neil, delegating to the upper execution context would make it easy to customize the behavior in different situations.